From 45f202bfa94ff18ca3a6918cddd05d533339fc6c Mon Sep 17 00:00:00 2001
From: rpitasky <111201305+rpitasky@users.noreply.github.com>
Date: Fri, 26 Aug 2022 19:55:02 -0400
Subject: [PATCH] 4553: Fix flags issue with z80 processor

Result flags are erroneously (see page 152 of the manual) not updated after performing an `adc a, (ix+d)` instruction.
---
 Ghidra/Processors/Z80/data/languages/z80.slaspec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Ghidra/Processors/Z80/data/languages/z80.slaspec b/Ghidra/Processors/Z80/data/languages/z80.slaspec
index 8f87c83fe..da940d6a1 100644
--- a/Ghidra/Processors/Z80/data/languages/z80.slaspec
+++ b/Ghidra/Processors/Z80/data/languages/z80.slaspec
@@ -764,6 +764,7 @@ cc2: "C"   is bits3_3=0x7   { c:1 = $(C_flag); export c; }
 :ADC A, ixMem8         is op0_8=0xdd; op0_8=0x8e; ixMem8 & A{
 	val:1 = ixMem8;
 	additionWithCarry(A, val, A);
+	setResultFlags(A);
 }
 
 :ADC A, iyMem8         is op0_8=0xfd; op0_8=0x8e; iyMem8 & A {
-- 
2.38.1

